### PR TITLE
DNN-10166 add IP Filter validation

### PR DESCRIPTION
**Pull request by tpluscode**
Added a check to validate the IP address and subnet mask which throws ArgumentException if either of them are not in correct format.

### CODE REVIEW

File: DNN Platform/Library/Entities/IPFilter/IPFilterController.cs [hunk 1]
```diff
@@ -256,5 +261,20 @@ public bool IsAllowableDeny(string currentIP, IPFilterInfo ipFilter)
         }
 
         #endregion
+
+        private static void AssertValidIPFilter(IPFilterInfo ipFilter)
+        {
+            IPAddress parsed;
+            if (IPAddress.TryParse(ipFilter.IPAddress, out parsed) == false)
+            {
+                throw new ArgumentException("IP address is not in correct format");
```
Comment (tpluscode) [inline DNN Platform/Library/Entities/IPFilter/IPFilterController.cs]: What is the best practice for adding localization? Maybe we don't want to localize this message at all?

File: DNN Platform/Library/Entities/IPFilter/IPFilterController.cs [hunk 2]
```diff
@@ -256,5 +261,20 @@ public bool IsAllowableDeny(string currentIP, IPFilterInfo ipFilter)
...
+            }
+
+            bool isIPRange = string.IsNullOrEmpty(ipFilter.SubnetMask);
+            if (isIPRange && IPAddress.TryParse(ipFilter.SubnetMask, out parsed) == false)
+            {
+                throw new ArgumentException("Subnet mask is not in correct format");
```
Comment (tpluscode) [inline DNN Platform/Library/Entities/IPFilter/IPFilterController.cs]: Ditto

Comment (zyhfish) [reply DNN Platform/Library/Entities/IPFilter/IPFilterController.cs]: Hi @tpluscode , for the localization, you can check DNN Platform\Library\Common\Requires.cs line 102 for example. the localization strings can saved in Website\App_GlobalResources\Exceptions.resx.

**Comment by zyhfish** [comment]
Changes looks good and merged.

**Pull request merged**